src/h5.cxx

```diff
@@ -321,11 +323,17 @@ template <typename T> void swap(h5d_chunked_1d<T> &x, h5d_chunked_1d<T> &y) {
 template <typename T>
 append_ret h5d_chunked_1d<T>::append_data_1d(T const *data, hsize_t nlen) {
   append_ret ret{-1};
-  bool do_buf = nlen * sizeof(T) < 4 * 1024;
+  uint32_t nbytes = nlen * sizeof(T);
+  bool do_buf = nbytes <= buf_MAXPKG;
   if (do_buf) {
-    std::copy(data, data + nlen, std::back_inserter(buf));
+    auto p1 = (char *)data;
+    auto p2 = buf.data() + buf_n;
+    for (uint32_t i1 = 0; i1 < nbytes; ++i1) {
+      p2[i1] = p1[i1];
+    }
+    buf_n += nbytes;
   }
-  if (buf.size() > 128 * 1024 || (!do_buf && buf.size() > 0)) {
+  if (buf_n >= buf_SIZE - 2 * buf_MAXPKG || (!do_buf && buf_n > 0)) {
```

Comment:
What's the significance of `buf_SIZE - 2 * buf_MAXPKG`? Could you add a comment to clarify?